### PR TITLE
Fixed leaks

### PR DESCRIPTION
**Pull request by MayMeta**
Added 5 changes of Yuri Borovskih

### CODE REVIEW

File: audio.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def write_audio(audio_path, signal, sample_rate, mono = False):
 
 
 def resample(signal, sample_rate_, sample_rate):
-	return torch.as_tensor(librosa.resample(numpy.asarray(signal), sample_rate_, sample_rate)), sample_rate
+	return torch.from_numpy(np.expand_dims(librosa.resample(np.squeeze(signal.numpy()), sample_rate_, sample_rate), axis=0)), sample_rate
```
Comment (MayMeta) [inline audio.py:134]: Fix problem with librosa 0.8.0

File: train.py [hunk 2]
```diff
@@ -94,9 +94,9 @@ def evaluate_model(
 
 			for r in sum(stats, []) if args.verbose else []:
 				print(f'{val_dataset_name}@{iteration}:', batch_idx, '/', len(val_data_loader), '|', args.experiment_id)
-				print('REF: {labels} "{ref_}"'.format(ref_ = r['alignment']['ref'] if analyze else r['ref'], **r))
-				print('HYP: {labels} "{hyp_}"'.format(hyp_ = r['alignment']['hyp'] if analyze else r['hyp'], **r))
-				print('WER: {labels} {wer:.02%} | CER: {cer:.02%}\n'.format(**r))
+				print('REF: {labels_name} "{ref_}"'.format(ref_ = r['alignment']['ref'] if analyze else r['ref'], **r))
+				print('HYP: {labels_name} "{hyp_}"'.format(hyp_ = r['alignment']['hyp'] if analyze else r['hyp'], **r))
+				print('WER: {labels_name} {wer:.02%} | CER: {cer:.02%}\n'.format(**r))
```
Comment (MayMeta) [inline train.py:99]: r["labels"] doesn't exist anymore, now it's called r["labels_name"], according to vadimkantorov (metrics.py line 101)

File: datasets.py [hunk 3]
```diff
@@ -101,7 +102,7 @@ def __getitem__(self, index):
 			transcript = transcript[0]
 			signal, sample_rate = audio.read_audio(audio_path, sample_rate = self.sample_rate, mono = self.mono, backend = self.audio_backend, duration=self.max_duration) if self.frontend is None or self.frontend.read_audio else (audio_path, self.sample_rate)
 
-			transcript = dict(dict(audio_name = os.path.basename(transcript['audio_path'])), **transcript)
+			transcript = dict(dict(audio_name = os.path.basename(transcript['audio_path'])), ref = transcript['ref'], audio_path=transcript['audio_path'])
```
Comment (MayMeta) [inline datasets.py:105]: Mitigates the memory leaks problem

File: train.py [hunk 4]
```diff
@@ -584,6 +584,9 @@ def main(args):
 			if iteration and args.iterations and iteration >= args.iterations:
 				return
 
+			if args.iterations_per_epoch and iteration > 0 and iteration % args.iterations_per_epoch == 0:
+				break
+
```
Comment (MayMeta) [inline train.py:589]: Epochs are rounded to --iterations-per-epoch

File: vis.py [hunk 5]
```diff
@@ -521,7 +521,7 @@ def tabulate(
 		val_dataset_name = f[f.find('transcripts_') + len('transcripts_'):eidx]
 		checkpoint = os.path.join(experiment_dir, 'checkpoint_' + f[eidx:].replace('.json', '.pt')) if not json_ else f
 		metric = 'wer' if wer else 'entropy' if entropy else 'loss' if loss else 'per' if per else 'der' if der else 'cer'
-		val = torch.tensor([j[metric] for j in json.load(open(f)) if j['labels'].startswith(labels.alphabet)] or [0.0])
+		val = torch.tensor([j[metric] for j in json.load(open(f)) if j['labels_name'].startswith(labels.alphabet)] or [0.0])
```
Comment (MayMeta) [inline vis.py:524]: r["labels"] doesn't exist anymore, now it's called r["labels_name"], according to vadimkantorov (metrics.py line 101)
